tests/slicing/test-runner.py

```diff
@@ -211,6 +228,12 @@ def run_test(test, bccode, optafter, linkafter, args):
 
     set_environment()
 
+    # create tmpdir for given test
+    from os import mkdir
+    rmtree(argv[1], ignore_errors=True) # just to be sure it does not exist
+    mkdir(argv[1])
+    chdir(argv[1])
```

Comment:
This breaks running `make check` for in-source builds.

Reply:
Temporary folders for given tests are now generated by `mkdtemp`, so it should by okay now.

Reply:
The problem was not in the way the working directory was created (in fact, I quite liked that the directory had the name of the test) but that the paths got broken. This problem still persists:

```
$ tests/slicing/test-runner-debug.py test1
> clang  -include  sources/../test_assert.h  -emit-llvm  -c  sources/test1.c  -o  test1.bc
clang-10: error: no such file or directory: 'sources/test1.c'
clang-10: error: no input files
Failed executing clang
```

`clang` cannot find `sources/tests1.c` because it is now in the temporary subdirectory, but `SOURCESDIR` variable was not correctly adjusted.

Reply:
Thanks for the reproducer! It should be fixed now. I didn't realise you can run `test-runner.py` directly as in-source `make check` was was working for me without this fix already.